[develop]: New smoke verification features, other improvements

mkavulich opened the pull request:
## DESCRIPTION OF CHANGES: 

### New features

#### Pointstat and Gridstat tasks now parallelized, exscript converted to python

Pointstat and Gridstat tasks can now be parallelized across lead times. If more than one lead time is to be verified, the lead times will be split up among a number of tasks specified by the new `verification: VX_TASKS` setting. Each individual task will have its own METplus conf file with the suffix `.tasknum`, where `tasknum` is just the integer task number, as well as its own log file in the same vein.

This parallelization required converting the exscrtipt `scripts/exregional_run_met_gridstat_or_pointstat_vx.sh` to a python script `scripts/exregional_run_met_gridstat_or_pointstat.py`. This change should be invisible to the end user, but boy does it come with some benefits for developers, even beyond the ability to parallelize:

 - General speedup for the script, even when not using more than one task
 - Better settings control: variables in var_defns.yaml are retained as a dictionary rather than being collapsed into simple environment variables, so they can be referenced by their specific section from the config file
 - Better error handling with proper python tracebacks

#### Verification of smoke variables for HRRR-smoke
Previous additions of smoke variable verification (PM and AOD) had hard-coded the variable names for RRFS-specific output. Now the option exists to verify HRRR-smoke as well (PM2.5 and AOD only), using `FCST_SMOKE_TYPE: HRRR` (default remains `RRFS`)

#### New data retrieval protocol: awscli

There is a new protocol for data retrieval introduced. This "awscli" data retrieval method invokes the AWS command line interface to retrieve data from a specified AWS S3 data bucket. The main advantage of this method, aside from potentially being faster than wget, is that it allows for wildcards in filenames and the S3 directory structure, which is not possible via wget. The AWS CLI is installed via conda, and so does require an additional conda package, which is built at build-time and loaded in the "get_data" script.

#### Specify the verification domain in config.yaml, including custom domains

Previously, the verification domain for any particular task was hard-coded in the METplus config file templates; typically to CONUS but sometimes to FULL or both. Now, all verification tasks will verify the FULL domain always, and the various preset domains available in METplus can be specified in `config.yaml` with the `verification: VX_MASK` option. Multiple verification domains can be selected, and all verification statistics will be written to the same `.stat` files as they were previously.

In addition, users can now specify their own custom verification domain by creating a `.poly` file in the `parm/metplus/` directory, and adding that to. These `.poly` files are simple text files with pairs of lat/lon points, used by METplus to specify a specific closed-path domain on the globe where verification will take place. Two example files, `parm/metplus/AZCA.poly` and `parm/metplus/EAST_OF_ROCKIES.poly` are provided.

Finally, a plotting script that will plot the lat/lon points from these `.poly` files (`parm/metplus/plot_latlon_points.py`) as a `.png` image file is added.

#### Generalize `ush/mrms_pull_topofhour.py` to work with any observation type, renamed to `ush/select_validtime_obs.py`

The script `mrms_pull_topofhour.py` was intended to take MRMS observation files, which are valid at somewhat random times every few minutes, and select the file closest to the intended valid time, and rename it accordingly for later use in the verification system. Because the new, partially implemented `GOESAOD` and `GOESADP` observation types suffer the same problem, this script has been generalized to work with any observation type that has this property, and renamed accordingly to the more generic `ush/select_validtime_obs.py`.

### Improvements/bug fixes for verification
 - Fix AERONET observation retrieval so it only pulls data for the selected day, and data pulled from HTTP is formatted correctly
 - Rename observation data sources for simplicity
 - LOOP_ORDER setting moved into `common.conf` for simplicity
 - Some WE2E tests are updated to test new options `VX_MASK` and `VX_TASKS`
 - Old, unused script `ush/get_mrms_files.sh` is removed
 - 4 OpenMP tasks are now used for some parallel-capable MET tasks (`GenEnsProd` and `GridStat`)

### Miscellaneous improvements
 - A new `--delay` option is added for WE2E testing scripts, which specifies a delay in seconds between calls to `rocotorun`. Previously this was hard-coded to 60 to overcome a problem with Derecho; now it only defaults to 60 on Derecho specifically, and other machines default to 5. That way testing on other machines can stop being delayed unnecessarily.
 - New data location for newer HRRR data added
 - The Rocoto `taskthrottle` parameter, which limits the number of tasks that will be submitted at once, had been included in `ush/config_defaults.yaml` as `platform: TASKTHROTTLE` but wasn't actually hooked up to anything. This is now fully implemented, as well as the analogous `cyclethrottle` parameter.
 - Some defaults with environment variables are converted to use UWtools references to other variables that are resolved at experiment generation. This is done because config settings with environment variables do not work with python exscripts.

### Other changes
Two new observation types, `GOESAOD` and `GOESADP`, can now be retrieved, but are not yet used for anything.

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## TESTS CONDUCTED: 
These changes were used for several experiments verifying RRFS smoke and HRRR smoke forecasts, so the smoke-specific verification features are extensively tested. In addition, a fundamental test suite was run and passed for Hera, Intel, as well as all verification WE2E tests.

- [ ] derecho.intel
- [ ] gaeac6.intel
- [ ] hera.gnu
- [x] hera.intel
- [ ] hercules.intel
- [ ] orion.intel
- [ ] ursa.gnu
- [ ] ursa.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [x] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:
None

## DOCUMENTATION:
  - Updated the Users Guide to include all the new config.yaml options, as well as updating the tech doc for the renamed python script.


## ISSUE: 
None

## CHECKLIST
<!-- Add an X to check off a box. -->
- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

mkavulich committed:
Local changes for AMS work, including pygraf in workflow

mkavulich committed:
Add new conda environment with awscli tools

mkavulich committed:
Remove old, unused file get_mrms_files.sh, a relic from original RRFS merge process

mkavulich committed:
Pygraf modulefile for Jet

mkavulich committed:
Revert change to metplus version not needed, comment that pygraf is not only Hera

mkavulich committed:
Correct full RRFS archive locations, add warning message in retrieve_data.py if not all info has been provided, fix empty "success" message on failure

mkavulich committed:
Remove old custom vx mask file

mkavulich committed:
Rename GOES_AOD to GOESABI for generality, start adding necessary variables to config_defaults.yaml

mkavulich committed:
Update setup.py with missed checks for smoke/dust obs

mkavulich committed:
- Add separate entries for GOES_AOD and GOES_ABI due to limitations in how retrieve_data decides if all files were retrieved
 - Add check for "command not found" error for AWS CLI in retrieve_data; print more detailed error message if so

mkavulich committed:
- Rename GOES obs to "GOESAOD" and "GOESADP" for consistency
 - Add "get_obs" tasks for new obs; not working yet, but got through a significant chunk of get_obs.py successfully
 - Remove some duplicate checks for valid ob types
 - Consolidate some duplicate logic in get_obs.py
- Some logic consolidation in set_cycle_and_obs_timeinfo.py, remove unused imports

mkavulich committed:
- Remove "_obs" suffix from data_locations definitions
 - Add new "get_data" environment to get_obs modulefile for testing on Jet (will add to rest later)
 - Remove unused OBS_DIR variables from get_obs task definitions
 - Actually need a separate OBS_DIR for GOESADP unfortunately with current code

mkavulich committed:
Obs retrieval now fully working for GOES AOD and ADP :)

 - Refactor "select_validtime_obs" to remove hard-coded assumptions about file type and format
   - Remove "product" and "level" arguments
   - Introduce templates for specifying input and output filenames
   - Consolidate date-sorting logic
   - Remove remaining MRMS-specific references

mkavulich committed:
Changes from Michelle for NAM verification

mkavulich committed:
Add script to plot METplus .poly files

mkavulich committed:
Major changes to METplus: user-specifiable MASK regions, OMP_NUM_THREADS=4 for parallelized code, general cleanup

 - Always verify on FULL grid
 - Allow users to specify MASK_GRID regions in the config file, using the names of the grid as specified in the METplus users guide (or using custom .poly files in parm/metplus/)
 - Some general conf file cleanup
 - Allow user to specify input file for poly plotting script
 - Fix GOES task names in setup.py
 - New GOES obs in smoke test, test for smoke vx at 12z init; not sure if I'll keep these changes but using them for now

mkavulich committed:
- Add new observation type for data pulling (GOES_AOD)
 - Since GOES_AOD is only available from AWS (and google cloud, but let's not get into that),
   and filenames are not exactly known, we can't use wget to retrieve files, so this commit
   implements a new prototcol (awscli) that introduces the AWS command-line argument tool to
   retrieve_data.py.
 - To remove ambiguity, the "download" protocol is more accurately renamed to "wget"

mkavulich committed:
Fixes to METplus changes
 - Add forgotten AZCA.poly file
 - Fix error in mask specification in GridStat_ensmean.conf
 - Fix logic for populating VX_MASK files in conf files

mkavulich committed:
- Rename mrms_pull_topofhour to select_validtime_obs in preparation for use with GOES obs as well as MRMS
 - Rearrange logic for MRMS file naming and remove some unnecessarily convoluted logic
 - Handle unzipping of MRMS files in get_obs rather than select_validtime_obs

Tested MRMS REFC with this commit to ensure the reorganization was correct for existing obs

mkavulich committed:
Fixes after rebase

 - Fix list in experiment.jsonschema
 - Remove old rocoto: section from config_defaults.yaml

mkavulich committed:
Simplifying verification confs a bit: remove unusedoutput_base, move LOOP_ORDER to common.conf since this will always = times

mkavulich committed:
Various fixes after more testing

 - Fix GridStat ensmean task; conf file was missing "FULL" domain
 - Remove manual specification of OMP_NUM_THREADS in exscripts
 - Fix test MET_verification_smoke_12z_only_vx for new taskgroup format
 - Move some logic in setup.py so build.yaml is only read for tasks using compiled executables; in
   other words, verificaiton-only experiments don't need it

mkavulich committed:
Pythonize Gridstat/Pointstat exscript in preparation for parallelization across forecast hours

 - This required some messy logic to handle environment variables in paths; this can be handled
   better if everything is in python but for now it's a bit convoluted. To make this simpler I
   removed environment variables from paths where possible
 - Also required slight change to how jinja macros are loaded: they need to be loaded from the
   current directory rather than a full path. This actually makes things a little simpler overall
 - This does require calling run_metplus.py as a subprocess, which is not ideal. In the future
   it would be really helpful to import and call run_metplus as a function.

mkavulich committed:
Implement parallelization for GridStat and PointStat deterministic tasks

 - Adds new VX_TASKS variable for specifying the number of tasks to run in parallel for these tasks
 - Uses Python multiprocessing to split the forecast lead hours among the specified number of parallel tasks
 - Changes include automatic scaling of defaults for memory and nodes based on the VX_TASKS setting

mkavulich committed:
Various fixes

 - More correct name for runtime settings for PCPCombine fcst tasks
 - Fix problem with parallel conf file names, separate parallel tasks to different log files
 - Remove some unneeded prints, convert some to debug prints
 - Fix order of operations issue in setup.py that incorrectly handled differently capitalized machine name

mkavulich committed:
Fixes from Spring 2024 run:

 - Use polymask regions for Pointstat
 - Include changes to neighborhood width for GridStat
 - Specify CONUS vx only for some WE2E tests

mkavulich committed:
Fixes from 2024 Winter run:

 - Fix walltime specification for PcpCombine_APCP, make order consistent
 - Fix naming of processor-specific conf files for parallel METplus
 - Longer pause between calls to rocotorun to avoid overloading filesystem

mkavulich committed:
Fix aeronet obs pulling from HTTP:

 - Only retrieve day of interest for given experiment (previously we retrieved way too many obs)
 - Fix automatic renaming from strange web query format
 - Automatically remove HTML tags and fix header

mkavulich committed:
Make TASKTHROTTLE config setting actually work; add analogous CYCLETHROTTLE setting as well

mkavulich committed:
Mods for HRRR-smoke verification

mkavulich committed:
Implement fewer parallel tasks for UPA Pointstat since it uses way more memory

mkavulich committed:
RRFS PM10 fields are actually PM10 - PM25...need to add the latter back in in PcpCombine

mkavulich committed:
Add new "delay" argument to monitor_jobs.py to specify delay between calls to rocotorun (calling too frequently for very large experiments can cause issues)

mkavulich committed:
pygraf settings for RRFS smoke

mkavulich committed:
Its ridiculous to slow down experiments on every platform for problem on one platform. Already introduced "delay" as an argument for test monitoring; use this to control delay and only make it one minute on Derecho

mkavulich committed:
Remove pygraf stuff, we will not be merging this back to SRW

mkavulich committed:
Pre-PR cleanup

 - Remove features not intended for the public app (pygraf interface, RRFS prototype locations, some test settings)
 - Remove old, replaced bash script scripts/exregional_run_met_gridstat_or_pointstat_vx.sh
 - Some formatting fixes/remove debugging things

mkavulich committed:
Fix WE2E scripts

mkavulich committed:
More fix for WE2E test scripts, restore RRFS to pulling conus data

MichaelLueken commented on data_environment.yml:
```diff
@@ -0,0 +1,9 @@
+name: srw_app
```
Should the name associated with this be get_data?

```suggestion
name: get_data
```

MichaelLueken commented on modulefiles/tasks/jet/get_obs.local.lua:
Should the rest of the `get_obs.local.lua` task modulefiles be updated in this manner as well?

MichaelLueken commented on parm/metplus/plot_latlon_points.py:
```diff
@@ -0,0 +1,85 @@
+    parser.add_argument('-i', '--input_file', default='latlon.txt',
+                        help='Name of METplus polyfile')
+#    parser.add_argument('-o', '--output_file', default='map.png',
+#                        help='Name of output file')
```
Will this feature be uncommented in the future?  If so, then we can leave it in.  If this feature won't be exercised, it should be removed.

mkavulich replied:
I went ahead and removed this, since output files are named according to the corresponding input files

MichaelLueken commented on tests/WE2E/test_configs/verification/config.MET_verification_only_vx.yaml:
```diff
@@ -22,5 +22,6 @@ workflow:
     - parm/wflow/verify_det.yaml
 
 verification:
+  VX_MASK: ["CONUS","EAST","EAST_ROCKIES"]
```
Should be `EAST_OF_ROCKIES`, rather than `EAST_ROCKIES`.

```suggestion
  VX_MASK: ["CONUS","EAST","EAST_OF_ROCKIES"]
```

MichaelLueken commented on tests/WE2E/monitor_jobs.py:
```diff
@@ -17,14 +17,15 @@
 
 def monitor_jobs(expts_dict: dict, monitor_file: str = '', procs: int = 1,
-                 mode: str = 'continuous', debug: bool = False) -> str:
+                 mode: str = 'continuous', delay: int = 10, debug: bool = False) -> str:
```
`delay` is set to `5` by default in `utils.py`.  Is there a reason for setting it to `10` here?

```suggestion
                 mode: str = 'continuous', delay: int = 5, debug: bool = False) -> str:
```

MichaelLueken commented on tests/WE2E/monitor_jobs.py:
```diff
@@ -180,6 +179,8 @@ def setup_logging(logfile: str = "log.run_WE2E_tests", debug: bool = False) -> N
                         help='YAML-format file specifying the information of jobs to be run; '\
                              'for an example file, see monitor_jobs.yaml', required=True)
+    parser.add_argument('--delay', type=int,
+                        help='Pause this number of seconds between calls to rocotorun', default=10)
```
Same as above, default for `delay` should be `5`.

```suggestion
    parser.add_argument('--delay', type=int,
                        help='Pause this number of seconds between calls to rocotorun', default=5)
```

MichaelLueken reviewed (commented):
@mkavulich -

I've completed my initial review of the changes that you have made and ran the verification WE2E tests on Ursa.

After making the minor change in `tests/WE2E/test_configs/verification/config.MET_verification_only_vx.yaml` (noted in this review), I was able to get all of the verification WE2E tests to successfully pass.

I've also left suggested changes and questions in a few other files, but these changes are looking good.  I'll wait until documentation is complete for my final review.

MichaelLueken commented on tests/WE2E/run_we2e_tests.py:
```diff
@@ -40,14 +40,21 @@ def run_we2e_tests(homedir, args) -> None:
 
+    # Derecho requires long delay between calls to rocotorun due to system-level cacheing of
+    # job statuses
+    if machine=="DERECHO":
```
On line 49, `machine` is set to `args.machine.lower()`.  Since `machine` is converted to lower case, should `DERECHO` be `derecho`?

```suggestion
    if machine=="derecho":
```

mkavulich committed:
Updating users guide for new options

mkavulich committed:
Fix renamed VX domain in WE2E test

mkavulich committed:
Apply suggestions from code review

Co-authored-by: Michael Lueken <63728921+MichaelLueken@users.noreply.github.com>

mkavulich committed:
Remove output file name from plot_latlon_points.py; files are automatically named based on the input polyfile

mkavulich committed:
Update tech docs

mkavulich commented:
@MichaelLueken Thanks for your review, I have addressed all of your initial comments.

I have gone ahead and updated the Users Guide to include all the new ``config.yaml`` options, as well as updating the tech doc for the renamed python script. Let me know if there's anything else you need from my end.

MichaelLueken commented:
The verification WE2E tests were run on Derecho and all successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
MET_ensemble_verification_20251023065459                           COMPLETE              46.41
MET_ensemble_verification_winter_wx_20251023065531                 COMPLETE             215.83
MET_ensemble_verification_only_vx_20251023065606                   COMPLETE               2.38
MET_verification_only_vx_20251023065630                            COMPLETE               0.48
MET_verification_20251023065649                                    COMPLETE              21.72
MET_verification_smoke_only_vx_20251023065718                      COMPLETE               0.45
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             287.27
```
The comprehensive WE2E tests were run on Ursa Intel and all tests successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
2020_CAD_20251022162820                                            COMPLETE              46.76
2020_CAPE_20251022162833                                           COMPLETE              46.40
2019_hurricane_barry_20251022162846                                COMPLETE              45.91
2019_halloween_storm_20251022162900                                COMPLETE              45.55
2019_hurricane_lorenzo_20251022162915                              COMPLETE              46.61
2019_memorial_day_heat_wave_20251022162930                         COMPLETE              44.64
2020_denver_radiation_inversion_20251022162946                     COMPLETE              46.48
2020_easter_storm_20251022163003                                   COMPLETE              46.52
2020_jan_cold_blast_20251022163020                                 COMPLETE              46.35
community_20251022163038                                           COMPLETE              33.30
custom_ESGgrid_20251022163056                                      COMPLETE              22.59
custom_ESGgrid_Central_Asia_3km_20251022163114                     COMPLETE              51.55
custom_ESGgrid_Great_Lakes_snow_8km_20251022163133                 COMPLETE              15.46
custom_ESGgrid_IndianOcean_6km_20251022163204                      COMPLETE              18.72
custom_ESGgrid_NewZealand_3km_20251022163224                       COMPLETE              77.13
custom_ESGgrid_Peru_12km_20251022163244                            COMPLETE              28.07
custom_ESGgrid_SF_1p1km_20251022163305                             COMPLETE             259.55
deactivate_tasks_20251022163326                                    COMPLETE               0.95
get_from_AWS_ics_GEFS_lbcs_GEFS_fmt_grib2_2022040400_ensemble_2me  COMPLETE            1101.30
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2019061200_2025102  COMPLETE              10.03
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2019061200_202510  COMPLETE              13.18
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2021032018_202510  COMPLETE              13.12
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2022060112_48h_20  COMPLETE             128.73
get_from_HPSS_ics_GDAS_lbcs_GDAS_fmt_netcdf_2022040400_ensemble_2  COMPLETE             114.64
get_from_HPSS_ics_GSMGFS_lbcs_GSMGFS_20251022163613                COMPLETE              10.76
get_from_HPSS_ics_HRRR_lbcs_RAP_20251022163640                     COMPLETE              24.79
get_from_HPSS_ics_RAP_lbcs_RAP_20251022163708                      COMPLETE              29.61
get_from_NOMADS_ics_FV3GFS_lbcs_FV3GFS_20251022163737              COMPLETE              19.57
grid_RRFS_AK_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20251  COMPLETE             157.27
grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_20251022163837  COMPLETE             267.54
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP_20251022163  COMPLETE              41.36
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              44.72
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_2025102216  COMPLETE              41.08
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20251  COMPLETE              10.30
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              20.84
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              20.32
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP_20251022164  COMPLETE              63.44
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta_202  COMPLETE              31.17
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP_20251022164340  COMPLETE              15.80
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2_20251  COMPLETE              10.80
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2025102216452  COMPLETE              38.04
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_202510  COMPLETE             368.85
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson  COMPLETE             499.83
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_20251022  COMPLETE             507.39
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_20251022164  COMPLETE             557.26
grid_RRFS_CONUScompact_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_  COMPLETE              42.13
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR_20251022  COMPLETE              38.44
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_RRFS_sas_2025  COMPLETE              38.55
grid_RRFS_CONUScompact_25km_ics_RRFS_lbcs_RRFS_suite_RRFS_sas_202  COMPLETE              17.18
grid_RRFS_CONUScompact_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_  COMPLETE              16.05
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR_2025102  COMPLETE              53.99
grid_RRFS_CONUScompact_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_2  COMPLETE             458.69
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR_202510221  COMPLETE             489.03
grid_RRFS_NA_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP_20251022165505  COMPLETE             106.76
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0_202510  COMPLETE              42.31
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_HRRR_2025102216570  COMPLETE              29.20
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0_20251022165  COMPLETE              23.14
grid_SUBCONUS_Ind_3km_ics_NAM_lbcs_NAM_suite_GFS_v16_202510221658  COMPLETE              36.27
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_RRFS_sas_plot_2025  COMPLETE              14.06
long_fcst_20251022170017                                           COMPLETE             131.81
MET_ensemble_verification_only_vx_20251022170105                   COMPLETE               1.08
MET_ensemble_verification_only_vx_time_lag_20251022170220          COMPLETE               3.73
MET_ensemble_verification_winter_wx_20251022170336                 COMPLETE             188.27
MET_verification_only_vx_20251022170529                            COMPLETE               0.16
pregen_grid_orog_sfc_climo_20251022170625                          COMPLETE              10.48
smoke_dust_grid_RRFS_CONUS_25km_suite_RRFS_sas_20251022170713      COMPLETE              67.75
specify_EXTRN_MDL_SYSBASEDIR_ICS_LBCS_20251022170810               COMPLETE              10.37
specify_template_filenames_20251022170904                          COMPLETE              11.38
vx-det_long-fcst_custom-vx-config_aiml-fourcastnet_20251022170958  COMPLETE               0.45
vx-det_long-fcst_custom-vx-config_aiml-graphcast_20251022171050    COMPLETE               0.43
vx-det_long-fcst_custom-vx-config_aiml-panguweather_2025102217114  COMPLETE               0.44
vx-det_long-fcst_custom-vx-config_gfs_20251022171238               COMPLETE               0.56
vx-det_long-fcst_winter-wx_SRW-staged_20251022171333               COMPLETE               2.18
vx-det_multicyc_fcst-overlap_ncep-hrrr_20251022171440              COMPLETE               5.27
vx-det_multicyc_first-obs-00z_ncep-hrrr_20251022171544             COMPLETE               2.96
vx-det_multicyc_last-obs-00z_ncep-hrrr_20251022171647              COMPLETE               4.04
vx-det_multicyc_long-fcst-no-overlap_nssl-mpas_20251022171750      COMPLETE              19.92
vx-det_multicyc_long-fcst-overlap_nssl-mpas_20251022171900         COMPLETE              15.79
vx-det_multicyc_no-00z-obs_nssl-mpas_20251022172008                COMPLETE               2.43
vx-det_multicyc_no-fcst-overlap_ncep-hrrr_20251022172113           COMPLETE               6.40
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            6975.98
```

MichaelLueken reviewed (approved):
@mkavulich -

Thank you very much for addressing my questions and comments!  The verification WE2E tests are successfully passing on Derecho and testing has completed on Ursa.  The documentation updates look good to me.

Approving now.

gsketefian commented:
@mkavulich FYI I just started reviewing this PR.  

One thing I noticed in the PR message is that in the section titled **Specify the verification domain in config.yaml, including custom domains**, the first sentence of the 2nd paragraph reads:
```
In addition, users can now specify their own custom verification domain by creating a .poly file in 
the parm/metplus/ directory, and adding that to.
```
I'm guessing there should be a path after the "to" at the end.  Can you fix that?  Thanks.

More comments and questions coming soon.

gsketefian commented on doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1667,17 +1670,17 @@ VX Parameters for Observations
    directory structure of each obs type.
 
-``[CCPA|NOHRSC|MRMS|NDAS|AERONET|AIRNOW]_OBS_AVAIL_INTVL_HRS``: (Defaults: [1|6|1|1|24|1])
+``[CCPA|NOHRSC|MRMS|NDAS|AERONET|AIRNOW|GOESAOD|GOESADP]_OBS_AVAIL_INTVL_HRS``: (Defaults: [1|6|1|1|24|1])
```
@mkavulich Update the `Defaults: [1|6|1|1|24|1]` to add the defaults for the two GOES types.

gsketefian commented on doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1667,17 +1670,17 @@ VX Parameters for Observations
 
-``[CCPA|NOHRSC|MRMS|NDAS|AERONET|AIRNOW]_OBS_DIR``: (Default: ``"{{ workflow.EXPTDIR }}/obs_data/[ccpa|nohrsc|mrms|ndas|aeronet|airnow]"``)
-   Base directory in which CCPA, NOHRSC, MRMS, NDAS, AERONET, or AIRNOW obs files needed by
+``[CCPA|NOHRSC|MRMS|NDAS|AERONET|AIRNOW|GOESAOD|GOESADP]_OBS_DIR``: (Default: ``"{{ workflow.EXPTDIR }}/obs_data/[ccpa|nohrsc|mrms|ndas|aeronet|airnow]"``)
```
@mkavulich Update the `Defaults: ...` here as well.

gsketefian replied:
@mkavulich Also, it would be nice to say somewhere (doesn't have to be here) what AOD and ADP stand for.  I know AOP, but I didn't know ADP stands for "Aerosol Detection Product" (according to google's AI).

mkavulich replied:
Added a description above

gsketefian replied:
Thanks!

gsketefian commented on doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1867,12 +1880,15 @@ VX Parameters for Observations
 
-``OBS_DATA_STORE_AIRNOW``: (Default: ``hpss``)
+``OBS_DATA_STORE_[CCPA|NOHRSC|MRMS|NDAS|AERONET|AIRNOW|GOESAOD|GOESADP]``: (Defaults: ``[hpss|hpss|hpss|hpss|hpss|hpss|aws|aws]``)
    Location to retrieve observation data from. Valid values are "aws" and/or "hpss", see
```
@mkavulich In the description of these variables, we say "Location ...".  That word makes me think of a url.  Maybe a better description would be "Data repository from which to retrieve observation files," or something similar.

gsketefian commented on jobs/JREGIONAL_RUN_MET_GRIDSTAT_OR_POINTSTAT_VX:
```diff
@@ -81,7 +81,7 @@ field (or group of fields).
 #-----------------------------------------------------------------------
 #
-$SCRIPTSdir/exregional_run_met_gridstat_or_pointstat_vx.sh || \
+python $SCRIPTSdir/exregional_run_met_gridstat_or_pointstat.py --debug || \
```
@mkavulich Should `--debug` be enabled as a default, or is it left over from debugging?

mkavulich replied:
I have included logic so it only uses the --debug flag when VERBOSE=true

gsketefian replied:
Ok.

gsketefian commented on parm/metplus/EnsembleStat.conf:
```diff
@@ -203,13 +192,7 @@ OBS_{{METPLUS_TOOL_NAME}}_WINDOW_END = {OBS_WINDOW_END}
 {{comment_or_null}}{{METPLUS_TOOL_NAME}}_CLIMO_CDF_BINS = 1
 {{comment_or_null}}{{METPLUS_TOOL_NAME}}_CLIMO_CDF_CENTER_BINS = False
 {{METPLUS_TOOL_NAME}}_CLIMO_CDF_WRITE_BINS = False
-{% if input_field_group in ['APCP', 'ASNOW'] %}
```
@mkavulich Couple of METplus questions that I think are straightforward but difficult to find answers to in the METplus or MET User's Guides:
1) If `{{METPLUS_TOOL_NAME}}_MASK_GRID` is not specified in this conf file or if it's set to nothing, I assume a mask is not used, i.e. every geographic point available in the data is included in the verification.  Is that correct?
2) Does setting `{{METPLUS_TOOL_NAME}}_MASK_GRID` to `FULL` mean the same thing, i.e. don't use a mask?

mkavulich replied:
The `MASK_GRID`/`VERIFICATION_MASK` variables are kind of weird. They both kind of control the same thing but in different ways; from what I recall, `{{METPLUS_TOOL_NAME}}_MASK_GRID` defaults to FULL; if you set it to nothing, it will not verify *any* grid, and only verify masks specified by `{{METPLUS_TOOL_NAME}}_VERIFICATION_MASK`. It can also be used to verify a mask based on some other provided grid file or named NCEP grid name, but we haven't implemented that capability in the workflow. The `{{METPLUS_TOOL_NAME}}_VERIFICATION_MASK` variable controls what masks/subdomains defined by `.poly` files are verified on.

gsketefian replied:
I asked about this [here](https://github.com/dtcenter/METplus/discussions/3149).  Waiting for a bit more clarification to be sure.

gsketefian replied:
@mkavulich I looked into this some more, and it got messy!  
1. I found another variable in METplus that controls masking:  `{{METPLUS_TOOL_NAME}}_MASK_POLY`.  We're apparently using a deprecated versions of this without the `_MASK` substring in several places in our config templates (see [here](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-POINT_STAT_GRID)).  Here's what I found under `parm/metplus`:
    ```
    $ grep -n -r "POLY"
    PointStat_ensprob.conf:148:{{METPLUS_TOOL_NAME}}_POLY = {MET_INSTALL_DIR}/share/met/poly/CONUS.poly
    GridStat_or_PointStat.conf:151:{{METPLUS_TOOL_NAME}}_POLY = {{vx_mask}}
    Pb2nc_obs.conf:78:PB2NC_POLY =
    PointStat_ensmean.conf:148:{{METPLUS_TOOL_NAME}}_POLY = {MET_INSTALL_DIR}/share/met/poly/CONUS.poly
    ```
    It seems like we should insert `_MASK` before all the `_POLY`s at these four locations.

2. We should update the right-hand sides of the lines in `GridStat_or_PointStat.conf`, `Pb2nc_obs.conf`, and `PointStat_ensmean.conf` involving `[_MASK]_POLY` to use `{{vx_mask}}` instead of being hard-coded to nothing or CONUS.  

3. One of the other two masking variables, `{{METPLUS_TOOL_NAME}}_MASK_GRID` (described [here](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-GRID_STAT_MASK_GRID)), defines `mask.grid` in the MET configuration, which in turn is only for specifying NCEP grids/masks (see [here](https://metplus.readthedocs.io/projects/met/en/latest/Users_Guide/config_options.html#grid)).  We have this defined in 4 places:
    ````
    $ grep -n -r -i "MASK_GRID" *
    EnsembleStat.conf:195:{{METPLUS_TOOL_NAME}}_MASK_GRID = FULL
    GridStat_ensmean.conf:541:{{METPLUS_TOOL_NAME}}_MASK_GRID = FULL
    GridStat_ensprob.conf:525:{{METPLUS_TOOL_NAME}}_MASK_GRID = FULL
    GridStat_or_PointStat.conf:141:{{METPLUS_TOOL_NAME}}_MASK_GRID = FULL
    ````
    These are all hard-coded to `FULL`.  Do we want that, which I think means the full grid will always be verified in addition to any other masks specified in `VX_MASK`, or do we only want to verify what's in `VX_MASK`?  In the latter case, I think we have to set `{{METPLUS_TOOL_NAME}}_MASK_GRID` to a blank value, but I'm not 100% sure (documentation says "An empty list indicates that no masking grids should be used", but I'm not sure whether that means all points are thrown out or all points are used).
4. The third masking variable, `{{METPLUS_TOOL_NAME}_VERIFICATION_MASK_TEMPLATE`, is described [here](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-GRID_STAT_VERIFICATION_MASK_TEMPLATE).  It's not clear to me what dictionary/value that maps to in MET, but likely it also `mask.poly` (see [here](https://metplus.readthedocs.io/projects/met/en/latest/Users_Guide/config_options.html#poly)).  What do you think about changing all the `{{METPLUS_TOOL_NAME}}_VERIFICATION_MASK_TEMPLATE` variables in the config templates to `{{METPLUS_TOOL_NAME}}_MASK_POLY`?

gsketefian replied:
@mkavulich I will make an issue for this in the SRW repo, which we can then move into the new vx repo.

gsketefian commented on parm/metplus/plot_met_poly_points.py:
```diff
@@ -0,0 +1,83 @@
+import argparse
```
@mkavulich If this script is specifically for plotting only MET(plus) poly files, we should emphasize that by calling it something like `plot_met_poly_mask.py`.  Also it would be worth documenting this in the rst doc files so people know about it when creating new poly files (maybe you did but I missed it).

mkavulich replied:
I have renamed to `plot_met_poly_points.py` since it is only plotting the points in each file; also added a reference to it in the Users Guide.

gsketefian replied:
Ok, thanks.

gsketefian commented on parm/wflow/verify_ens.yaml:
```diff
@@ -43,6 +43,7 @@ metatask_GenEnsProd_EnsembleStat_APCP_all_accums:
       OBTYPE: 'CCPA'
       FCST_LEVEL: 'A#ACCUM_HH#'
       FCST_THRESH: 'all'
+      OMP_NUM_THREADS: 4
```
@mkavulich Is `OMP_NUM_THREADS` needed in the new parallelization that you implemented in this PR?

mkavulich replied:
No, this is simply an additional speedup for these tasks that we haven't been taking advantage of. I will add this to the PR description.

gsketefian replied:
Ok.

gsketefian commented on scripts/exregional_run_met_gridstat_or_pointstat.py:
```diff
@@ -0,0 +1,443 @@
+from set_leadhrs import set_leadhrs
+
+def set_vx_params(obtype,field_group,accum_hh):
+    """Function for reterning various verification parameters based on input args
```
reterning -> returning

gsketefian replied:
@mkavulich Can you fix this typo?  Thanks.

gsketefian commented on tests/WE2E/monitor_jobs.py:
```diff
@@ -17,14 +17,15 @@
 
 def monitor_jobs(expts_dict: dict, monitor_file: str = '', procs: int = 1,
-                 mode: str = 'continuous', debug: bool = False) -> str:
+                 mode: str = 'continuous', delay: int = 5, debug: bool = False) -> str:
```
@mkavulich I thought on Hera, the admins recommend 5 **minutes** between resubmitting jobs.  I usually use 1 minute, which seems to work fine, but I would have thought 5 seconds will overwhelm the system.  I guess not.

mkavulich replied:
Until recently there was no delay at all included between calls to rocotorun, and only 10 seconds between loops over experiments. Including a small delay seems prudent, but remember this isn't the time between job submissions, it's just the time between calls to rocotorun, which checks the status of prerequisite jobs before submitting any new ones.

The only platform we've seen issues on is Derecho, because admins use cacheing with qstat commands that mean that job information isn't always necessarily up-to-date; that was dealt with by a hard-coded delay in #1311, and I have retained that for Derecho below.

gsketefian replied:
Ok.

gsketefian commented on tests/WE2E/utils.py:
```diff
@@ -106,14 +106,14 @@ def print_WE2E_summary(expts_dict: dict, debug: bool = False):
     Args:
         expt_dir (str): Experiment directory name
-
+        delay        (int): [optional] Delay in seconds between calls to rocotorun.
```
Too many spaces between "delay" and "(int)".

gsketefian replied:
Thanks for fixing.

gsketefian commented on tests/WE2E/utils.py:
```diff
@@ -245,6 +245,7 @@ def update_expt_status(expt: dict, name: str, refresh: bool = False, debug: bool
         expt    (dict): A dictionary containing the information for an individual experiment, as described in the main ``monitor_jobs()`` function.
         name     (str): Name of the experiment; used for logging only
         refresh (bool): If True, this flag will check an experiment status even if it is listed as DEAD, ERROR, or COMPLETE. Used for initial checks for experiments that may have been restarted.
+        delay        (int): [optional] Delay in seconds between calls to rocotorun.
```
Too many spaces between "delay" and "(int)", I think.

gsketefian replied:
Thanks for fixing.

gsketefian commented on ush/experiment.jsonschema:
I realize this isn't directly related to your PR, but I noticed that in this file, on [these lines](https://github.com/ufs-community/ufs-srweather-app/pull/1339/files#diff-cab364125c82ace5bf9f7418640138d4a31ce8d8d5ed3278fa4813e218ad2a2eL33-L35), there is:
```
        "": {
          "type": "boolean"
        },
```
Does it make sense for the key (or whatever it's called) to the left of the colon be an empty string?  If so, what does that do?

gsketefian replied:
After looking carefully at the `global:` section in `config_defaults.yaml`, it looks like this section needs to contain the following booleans:
```
NEW_LSCALE
SPPT_LOGIT
SPPT_SFCLIMIT
```
We can make an issue for this if you don't want to make the change in this PR.

mkavulich replied:
Yes, please make a separate issue for this; honestly I don't quite understand the function of this typing because it doesn't seem to work all the time.

gsketefian replied:
@mkavulich I started a discussion [here](https://github.com/ufs-community/ufs-srweather-app/discussions/1342).

gsketefian commented on ush/setup.py:
```diff
@@ -75,6 +75,9 @@ def load_config_for_setup(ushdir, default_config_path, user_config_path):
+    user_config["user"]["MACHINE"] = machine
+
     # Check user config against experiment schema
     schema = ushdir / "user.jsonschema"
```
@mkavulich Again, not part of this PR, but wanted to point out I that the contents of `user.jsonschema` are already in `experiment.jsonschema`.  Is there a reason to repeat the contents like that?  Can we instead use `experiment.jsonschema` here?

gsketefian replied:
Searching online, seems that yes, we can.  But again, you don't have to fix it here.  We can make an issue.

mkavulich replied:
I don't really understand what `user.jsonschema` is for, is it possible it's just a relic of some previous work? It isn't mentioned in the documentation or the PR message for #1204 where it was introduced.

gsketefian replied:
@mkavulich I started a discussion [here](https://github.com/ufs-community/ufs-srweather-app/discussions/1343).

gsketefian commented on ush/setup.py:
```diff
@@ -75,6 +75,9 @@ def load_config_for_setup(ushdir, default_config_path, user_config_path):
     )
     logging.debug(user_config)
 
+    machine = user_config["user"]["MACHINE"].upper()
```
@mkavulich Is it possible to do this on one line, like:
```
    user_config["user"]["MACHINE"] = user_config["user"]["MACHINE"].upper()
```

mkavulich replied:
The assignment is for both `machine` (local variable) and `user_config["user"]["MACHINE"]` (config variable), so two different assignments are necessary. This is previously existing logic that was moved up.

gsketefian replied:
Ok, makes sense.

mkavulich committed:
Suggestions from Gerards review

mkavulich committed:
Run python script verbose when VERBOSE is set in config

mkavulich committed:
More PR review comments addressed

mkavulich committed:
Rename new .poly plotting script, include reference in users guide

mkavulich committed:
One more doc update

mkavulich reviewed (commented):
@gsketefian I think I've addressed all your comments, let me know if there's anything else

MichaelLueken commented:
Due to the base wxvx conda environment being loaded on Gaea C6, the WE2E coverage tests fail to generate on that platform.

The Gaea C6 coverage WE2E tests successfully pass when manually running them outside of the role account:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_ESGgrid_NewZealand_3km_20251114103630                       COMPLETE              73.42
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_RRFS_sas_2025  COMPLETE              38.24
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_2025111410  COMPLETE              41.62
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson  COMPLETE             449.08
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR_2025111  COMPLETE              47.68
smoke_dust_grid_RRFS_CONUS_3km_suite_HRRR_gf_20251114103752        COMPLETE            1370.40
2020_CAPE_20251114103814                                           COMPLETE              44.03
2020_easter_storm_20251114103835                                   COMPLETE              43.58
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0_202511  COMPLETE              39.15
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            2147.20
```
```
Skill Score: 0.99485
+ [[ 0.99485 < 0.700 ]]
+ echo 'Congrats! You pass check!'
```

MichaelLueken commented:
Several WE2E tests on Ursa Intel are in STUCK status.

I have manually launched the coverage WE2E tests on Ursa Intel and all tests successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
2019_memorial_day_heat_wave_20251114160303                         COMPLETE              46.56
custom_ESGgrid_Peru_12km_20251114160325                            COMPLETE              28.10
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2019061200_2025111  COMPLETE              10.17
get_from_HPSS_ics_GDAS_lbcs_GDAS_fmt_netcdf_2022040400_ensemble_2  COMPLETE             116.02
get_from_HPSS_ics_HRRR_lbcs_RAP_20251114160413                     COMPLETE              24.88
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              20.71
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP_20251114160451  COMPLETE              16.05
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR_202511141  COMPLETE             492.11
MET_ensemble_verification_only_vx_time_lag_20251114160543          COMPLETE               3.80
pregen_grid_orog_sfc_climo_20251114160614                          COMPLETE              10.64
smoke_dust_grid_RRFS_CONUS_25km_suite_RRFS_sas_20251114160634      COMPLETE              66.96
vx-det_long-fcst_custom-vx-config_aiml-graphcast_20251114160657    COMPLETE               0.54
vx-det_multicyc_long-fcst-overlap_nssl-mpas_20251114160719         COMPLETE              10.67
vx-det_multicyc_long-fcst-no-overlap_nssl-mpas_20251114160745      COMPLETE              15.53
vx-det_multicyc_first-obs-00z_ncep-hrrr_20251114160813             COMPLETE               1.19
vx-det_multicyc_no-00z-obs_nssl-mpas_20251114160839                COMPLETE               1.19
vx-det_multicyc_no-fcst-overlap_ncep-hrrr_20251114160906           COMPLETE               3.01
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0_202511  COMPLETE              42.55
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             910.68
```
```
Skill Score: 0.9875
+ [[ 0.9875 < 0.700 ]]
+ echo 'Congrats! You pass check!'
```

MichaelLueken commented:
The UFS Fire WE2E tests were run on Gaea C6 and successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
UFS_FIRE_multifire_two-way-coupled_20251114120931                  COMPLETE              23.95
UFS_FIRE_one-way-coupled_20251114120944                            COMPLETE              23.33
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE              47.28
```
The AQM WE2E test also passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
aqm_grid_AQM_NA13km_suite_GFS_v16_20251114131727                   COMPLETE            4506.64
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            4506.64
```

MichaelLueken commented:
Several WE2E tests on Hera Intel are in STUCK status.

I have manually launched the coverage WE2E tests on Hera Intel and all tests successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
2019_memorial_day_heat_wave_20251114163917                         COMPLETE              65.06
custom_ESGgrid_Peru_12km_20251114163939                            COMPLETE              38.03
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2019061200_2025111  COMPLETE               9.73
get_from_HPSS_ics_GDAS_lbcs_GDAS_fmt_netcdf_2022040400_ensemble_2  COMPLETE             146.44
get_from_HPSS_ics_HRRR_lbcs_RAP_20251114164049                     COMPLETE              20.05
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              17.01
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP_20251114164137  COMPLETE              14.26
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR_202511141  COMPLETE             751.33
MET_ensemble_verification_only_vx_time_lag_20251114164241          COMPLETE               4.83
pregen_grid_orog_sfc_climo_20251114164319                          COMPLETE              10.02
smoke_dust_grid_RRFS_CONUS_25km_suite_RRFS_sas_20251114164344      COMPLETE              50.94
vx-det_long-fcst_custom-vx-config_aiml-graphcast_20251114164413    COMPLETE               0.57
vx-det_multicyc_long-fcst-overlap_nssl-mpas_20251114164440         COMPLETE              13.04
vx-det_multicyc_long-fcst-no-overlap_nssl-mpas_20251114164512      COMPLETE              16.75
vx-det_multicyc_first-obs-00z_ncep-hrrr_20251114164547             COMPLETE               1.10
vx-det_multicyc_no-00z-obs_nssl-mpas_20251114164621                COMPLETE               1.30
vx-det_multicyc_no-fcst-overlap_ncep-hrrr_20251114164655           COMPLETE               2.97
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0_202511  COMPLETE              43.88
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            1207.31
```
```
Skill Score: 0.9875
+ [[ 0.9875 < 0.700 ]]
+ echo 'Congrats! You pass check!'
```

MichaelLueken commented:
The Jenkins WE2E tests successfully passed on Derecho, Hera GNU, Hercules, Orion, and Ursa GNU.

Given the successful completion of manual coverage WE2E tests on Gaea C6, Hera Intel, and Hera GNU, shown in comments above, this PR is now ready to be merged.
